src/components/create-safe/useEstimateSafeCreationGas.ts

```diff
@@ -24,24 +22,21 @@ export const useEstimateSafeCreationGas = ({
   gasLimitLoading: boolean
 } => {
   const web3ReadOnly = useWeb3ReadOnly()
-  const chainId = useChainId()
-  const currentChain = useAppSelector((state) => selectChainById(state, chainId))
+  const chain = useCurrentChain()
   const wallet = useWallet()
 
-  const proxyContract = getProxyFactoryContractInstance(chainId)
-  const proxyContractAddress = proxyContract.getAddress()
-
-  const encodedSafeCreationTx = getSafeCreationTx({ owners, threshold, saltNonce, chain: currentChain })
-
   const [gasLimit, gasLimitError, gasLimitLoading] = useAsync<BigNumber>(() => {
-    if (!wallet?.address || !encodedSafeCreationTx || !web3ReadOnly) return
+    if (!wallet?.address || !chain || !web3ReadOnly) return
+
+    const proxyContract = getProxyFactoryContractInstance(chain.chainId)
```

Comment:
We should rename this to `proxyFactoryContract`. Proxy contract is the Safe instance itself.